plugins/modules/android_sdk.py

```diff
@@ -0,0 +1,213 @@
+    - Allows installation from different channels (stable, beta, dev, canary).
+    - Allows installation of packages to a non-default SDK root directory.
+author: Stanislav Shamilov (@shamilovstas)
+extends_documentation_fragment:
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+version_added: 10.2.0
+options:
+  accept_licenses:
+    description:
+      - If this is set to B(true), the module will try to accept license prompts generated by C(sdkmanager) during
+        package installation. Otherwise, every license prompt will be rejected.
+    type: bool
+    default: false
+  name:
+    description:
+      - A name of an Android SDK package (for instance, V(build-tools;34.0.0)).
+    aliases: ['package', 'pkg']
+    type: list
+    elements: str
+  state:
+    description:
+      - Indicates the desired package(s) state.
+      - V(present) ensures that package(s) is/are present.
+      - V(absent) ensures that package(s) is/are absent.
+      - V(latest) ensures that package(s) is/are installed and updated to the latest version(s).
+    choices: ['present', 'absent', 'latest']
+    default: present
+    type: str
+  sdk_root:
+    description:
+      - Provides path for an alternative directory to install Android SDK packages to. By default, all packages
+        are installed to the directory where C(sdkmanager) is installed.
+    type: path
+  channel:
+    description:
+      - Indicates what channel must C(sdkmanager) use for installation of packages.
+    choices: ['stable', 'beta', 'dev', 'canary']
+    default: stable
+    type: str
+requirements:
+  - C(java) >= 17
+  - C(sdkmanager) Command line tool for installing Android SDK packages.
+notes:
+  - For some of the packages installed by C(sdkmanager) is it necessary to accept licenses. Usually it is done through
+    command line prompt in a form of a Y/N question when a licensed package is requested to be installed. If there are
+    several packages requested for installation and at least two of them belong to different licenses, the C(sdkmanager)
+    tool will prompt for these licenses in a loop.
+    In order to install packages, the module must be able to answer these license prompts. Currently, it is only
+    possible to answer one license prompt at a time, meaning that instead of installing multiple packages as a single
+    invocation of the C(sdkmanager --install) command, it will be done by executing the command independently for each
+    package. This makes sure that at most only one license prompt will need to be answered.
+    At the time of writing this module, a C(sdkmanager)'s package may belong to at most one license type that needs to
+    be accepted. However, if this is changes in the future, the module may hang as there might be more prompts generated
+    by the C(sdkmanager) tool which the module won't be able to answer. If this is the case, file an issue and in the
```

Comment:
```suggestion
    by the C(sdkmanager) tool which the module will not be able to answer. If this is the case, file an issue and in the
```
(https://docs.ansible.com/ansible/devel/dev_guide/style_guide/grammar_punctuation.html#contractions - and yes, I'm aware of the irony that that page is using some contractions itself :) )